api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -57,7 +57,7 @@ public PositionOutputStream create() {
       }
 
       if (!file.getParentFile().isDirectory() && !file.getParentFile().mkdirs()) {
```

Comment:
Maybe we could just remove these checks.  I feel like this is pretty bad error handling since we check two different conditions and don't report which or why we couldn't create the directory.

What if instead we just allowed the underlying exception to propagate up?

Reply:
It looks like there are only four places not in test where it's being used and two or three of those are in catch blocks.  I think we could probably find a better set of exceptions to use in the few places where it is thrown and remove this entirely.

Reply:
I agree that there is no point in throwing the exception without mentioning why it has failed. 

I think we still need to create a directory for the parent path right? So, I kept that code and removed the exception part.